quickjs/src/nativeMain/kotlin/com/dokar/quickjs/QuickJs.native.kt

```
@@ -367,31 +404,52 @@ actual class QuickJs private constructor(
             // to a native interrupt.
             val interruptOnCancel = Job(coroutineContext.job)
             interruptOnCancel.invokeOnCompletion { cause ->
-                if (cause is CancellationException && !isClosed) {
-                    qjs_interrupt_request(interruptState)
+                if (cause is CancellationException) {
+                    requestInterruptIfOpen()
                 }
             }
             try {
                 evalInSession(EvaluationSession(), isRoot = true, block)
             } catch (e: QuickJsException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)